src/cpu/o3/inst_queue.cc

```diff
@@ -897,10 +897,22 @@ InstructionQueue::scheduleReadyInsts()
 
             listOrder.erase(order_it++);
             iqStats.statIssuedInstType[tid][op_class]++;
-        } else {
+        } else if (idx != FUPool::NoCapableFU) {
```

Comment:
IMHO it would be more intuitive to write this as

`idx == FUPool::NoFreeFU`

as it is easier to understand under which condition the code will be executed